### PR TITLE
[WIRE-422] Custom templates support

### PR DESCRIPTION
**Pull request by rjtokenring**
### Motivation
Support for custom template lifecycle

### Reviewer checklist

* [ ] PR address a single concern.
* [ ] PR title and description are properly filled.
* [ ] Changes will be merged in `main`.
* [ ] Changes are covered by tests.
* [ ] Logging is meaningful in case of troubleshooting.
* [ ] History is clean, commit messages are meaningful (see `CONTRIBUTING.md`) and are well formatted.

### CODE REVIEW

File: TT/resources/sketches/ToSiteWise 3/ToSiteWise_3_may08a.ino [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#include <ArduinoIoTCloud.h>
```
Comment (mirkokurt) [inline TT/resources/sketches/ToSiteWise 3/ToSiteWise_3_may08a.ino:1]: Did you intentionally put the sitewise template in the repo as example?

**Review by mirkokurt — APPROVED** [review]
Wow, this is a great work! Thank you for adding this feature.
Since this is an entirely new use case for the Cloud-cli, what do you think about adding a new section in the readme explaining how to handle custom templates?

File: TT/resources/sketches/ToSiteWise 3/ToSiteWise_3_may08a.ino (hunk 1 above)
Comment (rjtokenring) [reply TT/resources/sketches/ToSiteWise 3/ToSiteWise_3_may08a.ino]: No, it was my error... this is a test template used in manual test. Removed, thanks.

**Comment by rjtokenring** [comment]
> Wow, this is a great work! Thank you for adding this feature. Since this is an entirely new use case for the Cloud-cli, what do you think about adding a new section in the readme explaining how to handle custom templates?

Yes, will add

**Pull request merged**